internal/controllers/gateway/route_utils.go

```diff
@@ -82,6 +84,13 @@ const (
 	// https://github.com/kubernetes-sigs/gateway-api/pull/1516
 	// TODO: swap this out with upstream const when released.
 	RouteReasonNoMatchingParent gatewayv1beta1.RouteConditionReason = "NoMatchingParent"
+	// This reason is used with the "Accepted" condition when the Gateway has no
+	// compatible Listeners whose Port matches the route
+	// NOTE: This should probably be proposed upstream.
+	RouteReasonNoMatchingListenerPort gatewayv1beta1.RouteConditionReason = "NoMatchingListenerPort"
```

Comment:
Shouldn't we use `RouteReasonNoMatchingParent` (defined above) in this case? How does that differ from `RouteReasonNoMatchingParent`?

Reply:
It's not in `main` 😅 

> I think I did not add it

Github says you did :) 

https://github.com/Kong/kubernetes-ingress-controller/commit/510a8ef4c027860760d419008d7f0325ad120984#diff-538cabd873585a7f0cfe12f019e7b14b1945cb5c346265e4f73a026fcb166688R89-R95

All in all: if there are no references to this then let's delete it.

Reply:
Removed it now.
It seems that there was a conflict after #3181 merged, and that PR removed `RouteReasonNoMatchingListenerPort`, but I resolved the conflict by accepting both changes.